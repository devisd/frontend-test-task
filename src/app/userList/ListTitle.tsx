type Props = {
  title: string;
  className: string;
};

export const ListTitle: React.FC<Props> = ({ title, className }) => {
  return (
    <>
      <li>
        <h3 className={className}>{title}</h3>
      </li>
    </>
  );
};
