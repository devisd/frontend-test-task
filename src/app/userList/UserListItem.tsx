import Avatar from "@mui/material/Avatar";
import ListItem from "@mui/material/ListItem";
import ListItemAvatar from "@mui/material/ListItemAvatar";
import ListItemText from "@mui/material/ListItemText";
import { Buttons } from "./buttons/Buttons";
import { User } from "../../types/types";

import styles from "./../../assets/styles/UserList.module.css";

interface IProps {
  users: User[];
  onClick: (user: User, rating: number) => void;
}

export const UserListItem: React.FC<IProps> = ({ users, onClick }) => {
  return (
    <>
      {users.map((user) => (
        <ListItem key={user.id} className={styles.list__item}>
          <img
            alt="Remy Sharp"
            src={user.avatar}
            className={styles.list__avatar}
          />
          <ListItemText
            primary={`${user.first_name} ${user.last_name}`}
            secondary={`/ ${user.username} /`}
          />
          <Buttons user={user} onClick={onClick} />
        </ListItem>
      ))}
    </>
  );
};
