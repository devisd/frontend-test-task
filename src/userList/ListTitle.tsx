import ListItem from "@mui/material/ListItem";

type Props = {
  title: string;
};

export const ListTitle: React.FC<Props> = ({ title }) => {
  return (
    <>
      <ListItem>
        <h3>{title}</h3>
      </ListItem>
    </>
  );
};
