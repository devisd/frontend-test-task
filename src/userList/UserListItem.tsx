import Avatar from "@mui/material/Avatar";
import ListItem from "@mui/material/ListItem";
import ListItemAvatar from "@mui/material/ListItemAvatar";
import ListItemText from "@mui/material/ListItemText";
import { Buttons } from "../buttons/Buttons";
import { User } from "../types/types";

interface IProps {
  users: User[];
  onClick: (user: User, rating: number) => void;
}

export const UserListItem: React.FC<IProps> = ({ users, onClick }) => {
  return (
    <>
      {users.map((user) => (
        <ListItem key={user.id}>
          <ListItemAvatar>
            <Avatar alt="Remy Sharp" src={user.avatar} />
          </ListItemAvatar>
          <ListItemText
            style={{ display: "block" }}
            primary={`${user.first_name} ${user.last_name}`}
            secondary={`/ ${user.username} /`}
          />
          <Buttons user={user} onClick={onClick} />
        </ListItem>
      ))}
    </>
  );
};
