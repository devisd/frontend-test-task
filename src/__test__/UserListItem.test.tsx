import { render, screen } from "@testing-library/react";
import { UserListItem } from "../app/userList/UserListItem";

const data = [
  {
    id: 3586,
    first_name: "Georgeanna",
    last_name: "Lueilwitz",
    username: "georgeanna.lueilwitz",
    avatar: "https://robohash.org/cumquilabore.png?size=300x300&set=set1",
    rating: 0,
  },
  {
    id: 4712,
    first_name: "Patti",
    last_name: "Lemke",
    username: "patti.lemke",
    avatar:
      "https://robohash.org/eligendiadipiscitemporibus.png?size=300x300&set=set1",
    rating: 0,
  },
  {
    id: 5821,
    first_name: "Leanna",
    last_name: "White",
    username: "leanna.white",
    avatar: "https://robohash.org/adplaceataut.png?size=300x300&set=set1",
    rating: 0,
  },
];

const onClick = jest.fn();

describe("UserListItem Component", () => {
  it("UserListItem renders", () => {
    render(<UserListItem users={data} onClick={onClick} />);
    expect(screen.getByText(/Georgeanna/)).toBeInTheDocument();
  });
  it("UserListItem render without data", () => {
    render(<UserListItem users={[]} onClick={onClick} />);

    expect(screen.queryByRole("listitem")).toBeNull();
  });

  it("UserListItem snapshot", () => {
    expect(
      render(<UserListItem users={data} onClick={onClick} />)
    ).toMatchSnapshot();
  });
});
