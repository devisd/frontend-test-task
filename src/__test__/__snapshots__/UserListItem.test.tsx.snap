// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserListItem Component UserListItem snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding list__item css-1p823my-MuiListItem-root"
      >
        <img
          alt="Remy Sharp"
          class="list__avatar"
          src="https://robohash.org/cumquilabore.png?size=300x300&set=set1"
        />
        <div
          class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            Georgeanna Lueilwitz
          </span>
          <p
            class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
          >
            / georgeanna.lueilwitz /
          </p>
        </div>
        <div
          class="buttons"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedError MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedError MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-180lf2n-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            -
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <span
            class="buttons__value"
          >
            0
          </span>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSuccess MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedSuccess MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-sxix9q-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            +
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding list__item css-1p823my-MuiListItem-root"
      >
        <img
          alt="Remy Sharp"
          class="list__avatar"
          src="https://robohash.org/eligendiadipiscitemporibus.png?size=300x300&set=set1"
        />
        <div
          class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            Patti Lemke
          </span>
          <p
            class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
          >
            / patti.lemke /
          </p>
        </div>
        <div
          class="buttons"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedError MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedError MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-180lf2n-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            -
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <span
            class="buttons__value"
          >
            0
          </span>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSuccess MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedSuccess MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-sxix9q-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            +
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </li>
      <li
        class="MuiListItem-root MuiListItem-gutters MuiListItem-padding list__item css-1p823my-MuiListItem-root"
      >
        <img
          alt="Remy Sharp"
          class="list__avatar"
          src="https://robohash.org/adplaceataut.png?size=300x300&set=set1"
        />
        <div
          class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
        >
          <span
            class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
          >
            Leanna White
          </span>
          <p
            class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
          >
            / leanna.white /
          </p>
        </div>
        <div
          class="buttons"
        >
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedError MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedError MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-180lf2n-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            -
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <span
            class="buttons__value"
          >
            0
          </span>
          <button
            class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSuccess MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedSuccess MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-sxix9q-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            +
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </li>
    </div>
  </body>,
  "container": <div>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding list__item css-1p823my-MuiListItem-root"
    >
      <img
        alt="Remy Sharp"
        class="list__avatar"
        src="https://robohash.org/cumquilabore.png?size=300x300&set=set1"
      />
      <div
        class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
        >
          Georgeanna Lueilwitz
        </span>
        <p
          class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
        >
          / georgeanna.lueilwitz /
        </p>
      </div>
      <div
        class="buttons"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedError MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedError MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-180lf2n-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          -
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <span
          class="buttons__value"
        >
          0
        </span>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSuccess MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedSuccess MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-sxix9q-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          +
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding list__item css-1p823my-MuiListItem-root"
    >
      <img
        alt="Remy Sharp"
        class="list__avatar"
        src="https://robohash.org/eligendiadipiscitemporibus.png?size=300x300&set=set1"
      />
      <div
        class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
        >
          Patti Lemke
        </span>
        <p
          class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
        >
          / patti.lemke /
        </p>
      </div>
      <div
        class="buttons"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedError MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedError MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-180lf2n-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          -
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <span
          class="buttons__value"
        >
          0
        </span>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSuccess MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedSuccess MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-sxix9q-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          +
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </li>
    <li
      class="MuiListItem-root MuiListItem-gutters MuiListItem-padding list__item css-1p823my-MuiListItem-root"
    >
      <img
        alt="Remy Sharp"
        class="list__avatar"
        src="https://robohash.org/adplaceataut.png?size=300x300&set=set1"
      />
      <div
        class="MuiListItemText-root MuiListItemText-multiline css-konndc-MuiListItemText-root"
      >
        <span
          class="MuiTypography-root MuiTypography-body1 MuiListItemText-primary css-10hburv-MuiTypography-root"
        >
          Leanna White
        </span>
        <p
          class="MuiTypography-root MuiTypography-body2 MuiListItemText-secondary css-83ijpv-MuiTypography-root"
        >
          / leanna.white /
        </p>
      </div>
      <div
        class="buttons"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedError MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedError MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-180lf2n-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          -
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
        <span
          class="buttons__value"
        >
          0
        </span>
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-outlined MuiButton-outlinedSuccess MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButton-root MuiButton-outlined MuiButton-outlinedSuccess MuiButton-sizeMedium MuiButton-outlinedSizeMedium css-sxix9q-MuiButtonBase-root-MuiButton-root"
          tabindex="0"
          type="button"
        >
          +
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </button>
      </div>
    </li>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
